dockerfiles/parity-ci-linux/Dockerfile

```diff
@@ -32,8 +32,13 @@ RUN set -eux; \
 	apt-get install -y --no-install-recommends \
 # libssl-dev is needed for cargo-audit and sccache
 		g++ libssl-dev gcc libc6-dev \
-		make cmake libudev-dev ca-certificates \
-		git pkg-config curl time rhash; \
+		make libudev-dev ca-certificates \
+		git pkg-config curl time rhash software-properties-common apt-transport-https; \
+	apt-add-repository ppa:ubuntu-toolchain-r/test; \
+	apt-add-repository 'deb https://apt.kitware.com/ubuntu/ xenial main'; \
+	apt-get -y update; \
+	apt-get install -y  --allow-unauthenticated gcc-9 g++-9 cmake; \
```

Comment:
noooooo!!!!!! `--allow-unauthenticated` is a remote root exploit by design!

More seriously, if we are going to do this, we need to be absolutely certain that this image never gets its hands on anything sensitive (either regarding confidentiality or integrity).  Could we upgrade to a newer version of the OS instead?

Reply:
Another option is to install GCC from source, since the GNU project provides signed release tarballs.

Reply:
gcc-9 is available in debian [testing](https://packages.debian.org/bullseye/gcc-9). I suggest to use it from there instead from a unknown or potentially unmaintained third party.

Reply:
didn't see that it's ubuntu based, so instead I suggest the [disco](https://packages.ubuntu.com/disco/gcc-9) package.

Reply:
please see the latest changes